write-fonts/src/tables/variations.rs

```diff
@@ -126,19 +126,24 @@ impl PackedDeltas {
 
         /// compute the number of deltas in the next run, and whether they are i8s or not
         fn next_run_len(slice: &[i16]) -> (usize, bool) {
-            let (first, rest) = slice.split_first().expect("bounds checked before here");
-            debug_assert!(*first != 0);
-            let is_1_byte = in_i8_range(*first);
+            let first = *slice.first().expect("bounds checked before here");
+            debug_assert!(first != 0);
+            let is_1_byte = in_i8_range(first);
 
-            for (i, raw) in rest.iter().copied().enumerate().take(MAX_POINTS_PER_RUN) {
-                let two_zeros = raw == 0 && rest.get(i + 1) == Some(&0);
-                let different_enc_len = in_i8_range(raw) != is_1_byte;
+            let mut idx = 1;
+            while idx < MAX_POINTS_PER_RUN && idx < slice.len() {
+                let cur = slice[idx];
```

Comment:
Just a general note: I prefer to write these in terms of iteration wherever possible, because it often ends up letting the compiler generate better code. In the past I have seen code like this compiled into something where there is a bounds check on each array access, which is happening in a loop, whereas the iterating version generates code with a single bounds check up front.